webapp/channels/src/components/admin_console/multiselect_settings.tsx

```diff
@@ -26,67 +25,61 @@ interface Props {
     noResultText?: React.ReactNode;
 }
 
-interface State {
-    error: boolean;
-}
-
-export default class MultiSelectSetting extends React.PureComponent<
-Props,
-State
-> {
-    static defaultProps: Partial<Props> = {
-        disabled: false,
-    };
-
-    constructor(props: Props) {
-        super(props);
+const MultiSelectSetting: React.FC<Props> = ({
+    id,
+    values,
+    label,
+    selected,
+    onChange,
+    disabled = false,
+    setByEnv,
+    helpText,
+    noResultText,
+}) => {
+    const [error, setError] = useState(false);
 
-        this.state = {error: false};
-    }
+    const handleChange = useCallback((newValue: ValueType<Option>) => {
+        const updatedValues = newValue ? (newValue as Option[]).map((n) => 
+        n.value) : [];
 
-    handleChange = (newValue: ValueType<Option>) => {
-        const values = newValue ? (newValue as Option[]).map((n) => {
-            return n.value;
-        }) : [];
+        onChange(id, updatedValues);
+        setError(false);
+    }, [id, onChange]);
 
-        this.props.onChange(this.props.id, values);
-        this.setState({error: false});
-    };
-
-    calculateValue = () => {
-        return this.props.selected.reduce<Option[]>((values, item) => {
-            const found = this.props.values.find((e) => e.value === item);
+   const calculateValue = useCallback(() => {
+        return selected.reduce<Option[]>((result, item) => {
+            const found = values.find((e) => e.value === item);
             if (found) {
-                values.push(found);
+                result.push(found);
             }
-            return values;
+            return result;
         }, []);
-    };
+    }, [selected, values]);
 
-    getOptionLabel = ({text}: { text: string}) => text;
+    const getOptionLabel = useCallback(({text}: { text: string}) => text, []);
```

Comment:
Since this doesn't use anything inside the component, can be moved outside of the component, to avoid calling useCallback every render, and so on.

Reply:
understood, updated.